sdk/python/feast/infra/offline_stores/bigquery.py

```diff
@@ -102,8 +102,15 @@ def get_historical_features(
 
             assert isinstance(config.offline_store, BigQueryOfflineStoreConfig)
 
+            # We should create a new dataset if the dataset name was not overridden in the config
+            should_create_dataset = "dataset" not in config.offline_store.__fields_set__
```

Comment:
I don't 100% agree with this approach. What if the dataset already exists but its not set in the config? I think we should just manually check whether the dataset exists before we run the create command.

Reply:
That'll require an additional `bigquery.datasets.get` permission on the dataset in question

Reply:
So `get` permissions is too much?

Reply:
Also, its probably possible for us to get around the `bigquery.datasets.get` permission by trying an operation inside the dataset and seeing if it succeeds (and which exception is thrown).